src/Request/uapi-request.js

```diff
@@ -164,7 +167,12 @@ module.exports = function uapiRequest(
       .then(sendRequest)
       .then(parseResponse)
       .then(validateSOAP)
-      .then(parseFunction.bind(uParser))// TODO merge Hotels
+      .then((res) => {
+        const parser = parseFunction.bind(uParser);
+        return parser(res, {
```

Comment:
You have already passed auth.provider in this parser in line 62, why do you need to pass it once again, and can't handle provider in the parser itself?